packages/client/src/components/UploadForm.tsx

```diff
@@ -1,96 +1,158 @@
+};
 
-		const uploadStatus = await uploadToS3(body.data.presignedS3Url, blob);
-		if (!uploadStatus) {
-			console.error('Failed to upload to s3');
-			return false;
-		}
+export const UploadForm = () => {
+	const [status, setStatus] = useState<RequestStatus>(RequestStatus.Ready);
+	const [errorMessage, setErrorMessage] = useState<string | undefined>();
+	const { token } = useContext(AuthContext);
 
-		const sendMessageResponse = await authFetch('/api/transcribe-file', token, {
-			method: 'POST',
-			headers: {
-				'Content-Type': 'application/json',
-			},
-			body: JSON.stringify({
-				s3Key: body.data.s3Key,
-				fileName: file.name,
-			}),
-		});
-		const sendMessageSuccess = sendMessageResponse.status === 200;
-		if (!sendMessageSuccess) {
-			console.error('Failed to call transcribe-file');
-			return false;
-		}
-		return true;
-	};
+	if (!token) {
+		return (
+			<InfoMessage message={'Login required'} status={RequestStatus.Failed} />
+		);
+	}
+
+	if (status === RequestStatus.InProgress) {
+		return (
+			<InfoMessage
+				message={'Upload in progress...'}
+				status={RequestStatus.InProgress}
+			/>
+		);
+	}
+
+	if (status === RequestStatus.Failed) {
+		return (
+			<>
+				<InfoMessage
+					message={`${errorMessage ?? 'Upload failed'}`}
+					status={RequestStatus.Failed}
+				/>
+				<button
+					onClick={() => setStatus(RequestStatus.Ready)}
+					className="font-medium text-blue-600 underline dark:text-blue-500 hover:no-underline"
+				>
+					Click here
+				</button>{' '}
+				to try again
+			</>
+		);
+	}
+
+	if (status === RequestStatus.Success) {
+		return (
+			<p className="text-gray-500 dark:text-gray-400 pt-3">
+				Upload complete. Transcription in progress - check your email for the
+				completed transcript. The transcription process is typically shorter
+				than the length of the media file.{' '}
+				<button
+					onClick={() => setStatus(RequestStatus.Ready)}
+					className="font-medium text-blue-600 underline dark:text-blue-500 hover:no-underline"
+				>
+					Click here
+				</button>{' '}
+				to transcribe another file
+			</p>
+		);
+	}
 
 	const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
 		event.preventDefault();
+		setStatus(RequestStatus.InProgress);
 
 		const maybeFileInput = document.querySelector(
-			'input[name=file]',
+			'input[id=files]',
 		) as HTMLInputElement;
-		if (!maybeFileInput) {
+		if (
+			!maybeFileInput ||
+			!maybeFileInput.files ||
+			maybeFileInput.files.length === 0
+		) {
+			setErrorMessage(
+				'Invalid file input - did you select a file to transcribe?',
+			);
+			setStatus(RequestStatus.Failed);
 			return;
 		}
 
-		const success = await uploadFileAndTranscribe(maybeFileInput);
-		setStatus(success);
-		if (success) {
-			maybeFileInput.value = '';
+		for (const file of maybeFileInput.files) {
+			const result = await uploadFileAndTranscribe(file, token);
+			if (!result) {
+				setStatus(RequestStatus.Failed);
+				return;
```

Comment:
About the return statement, is it because if there are multiple files, the first file that failes to upload, we want to stop the upload?

Reply:
That is what was happening but I think it was a mistake -we should try the next file rather than just exiting